qt/python/mantidqt/widgets/sliceviewer/view.py

```diff
@@ -219,28 +232,45 @@ def plot_matrix(self, ws, **kwargs):
                                      interpolation='none',
                                      transpose=self.dimensions.transpose,
                                      norm=self.colorbar.get_norm(),
+                                     extent=old_extent,
                                      **kwargs)
-        self.image._resample_image()
+
         self.draw_plot()
 
     def clear_image(self):
         """Removes any image from the axes"""
         if self.image is not None:
+            if self.line_plots:
+                self.delete_line_plot_lines()
             self.image.remove()
             self.image = None
 
     def clear_figure(self):
         """Removes everything from the figure"""
+        if self.line_plots:
+            self.delete_line_plot_lines()
+            self.axx, self.axy = None, None
         self.image = None
         self._grid_on = False
         self.fig.clf()
+        self.ax = None
 
     def draw_plot(self):
         self.ax.set_title('')
         self.colorbar.set_mappable(self.image)
         self.colorbar.update_clim()
         self.mpl_toolbar.update()  # clear nav stack
-        self.clear_line_plots()
+        self.delete_line_plot_lines()
+        self.update_line_plot_labels()
+        self.canvas.draw_idle()
+
+    def reset_image_view_to_data_limits(self):
+        """Reset the image view limits to encompass all of the data"""
+        e1, e2, e3, e4 = self.image.get_extent()
+        # if self.dimensions.transpose:
```

Comment:
Should this commented-out code be here?

Reply:
Thanks for catching this. I meant to remove that after testing something.